pkg/driver/controller.go

```diff
@@ -389,3 +389,23 @@ func getNode(topo *csi.TopologyRequirement) (string, error) {
 	}
 	return "", nil
 }
+
+func (cs *controller) ControllerGetVolume(ctx context.Context, req *csi.ControllerGetVolumeRequest) (*csi.ControllerGetVolumeResponse, error) {
+
+	volume, err := utils.GetCStorVolume(req.GetVolumeId())
+	if err != nil {
+		return nil, err
+	}
+
+	logrus.Infof("Healthy state: %s Volume: %s", volume.Name, volume.Status.Phase)
+	return &csi.ControllerGetVolumeResponse{
+		Volume: &csi.Volume{
+			VolumeId:      volume.Name,
+			CapacityBytes: volume.Spec.Capacity.MilliValue(),
```

Comment:
Let's use current Volume capacity rather than desired capacity
```suggestion
			CapacityBytes: volume.Status.Capacity.MilliValue(),
```